pytorch_lightning/trainer/trainer.py

```diff
@@ -444,6 +444,11 @@ def __init__(
         self.signal_connector = SignalConnector(self)
         self.tuner = Tuner(self)
 
+        # UserWarning for max_epochs if not set
+
+        if self.max_epochs is None and max_steps == -1 and max_time is None:
+            warnings.warn("`max_epochs` is not set. By default it will run for 1000 epochs.")
```

Comment:
Let's use rank_zero_warn :) And might be worth creating a utility to parse the max_epochs, max_steps before passing them to the Loops.
This would be cleaner.

Reply:
@tchaton I have updated to the rank_zero_warn of type `UserWarning`. Also, if you can let me know what exact utility do we need to parse `max_epochs`, `max_steps` it'd be great, as I'm not sure I understood what you meant :")

Reply:
Hey @Rajathbharadwaj,

I meant something like this

```py
# rank_zero_warning of type UserWarning for max_epochs if not set
if self.max_epochs is None and max_steps == -1 and max_time is None:
    rank_zero_warn(
        "`max_epochs` is not set. By default it will run for 1000 epochs.",
        UserWarning,
    )

fit_loop = FitLoop(
    min_epochs=(1 if (min_epochs is None and min_steps is None and max_time is None) else min_epochs),
    max_epochs=(
    max_epochs if max_epochs is not None else (1000 if (max_steps == -1 and max_time is None) else -1)
),
)
```

could become

```py
min_epochs, max_epochs = _parse_max_epochs_and_steps(max_steps, max_epochs, max_time, min_steps, min_epochs)

# utilities/...
def _parse_max_epochs_and_steps(max_steps, max_epochs, max_time, min_steps, min_epochs):
    if max_epochs is None and max_steps == -1 and max_time is None:
        rank_zero_warn(
            "`max_epochs` is not set. By default it will run for 1000 epochs.",
            UserWarning,
        )
    min_epochs = (1 if (min_epochs is None and min_steps is None and max_time is None) else min_epochs)
    max_epochs = max_epochs if max_epochs is not None else (1000 if (max_steps == -1 and max_time is None) else -1)
    return min_epochs, max_epochs
```

Reply:
Oh, okay, gotcha! thanks for the help @tchaton

Reply:
maybe pytorch_lightning/loops/utilities.py might be the most appropriate.

Reply:
Thanks @tchaton 
I've added the necessary changes mentioned by you please checkout! 
Again, thanks for helping me out!